src/language-markdown/parser-markdown.js

```diff
@@ -145,4 +146,25 @@ function frontmatter() {
   tokenizer.onlyAtStart = true;
 }
 
+function liquid() {
+  const proto = this.Parser.prototype;
+  const methods = proto.inlineMethods;
+  methods.splice(methods.indexOf("text"), 0, "liquid");
+  proto.inlineTokenizers.liquid = tokenizer;
+
+  function tokenizer(eat, value) {
+    const match = /^({%-?[\s\S]*?-?%}|{{-?[\s\S]*?-?}})/.exec(value);
```

Comment:
I think you're right about `-?`. I don't understand what you mean about the leading/trailing spaces.

I don't know if one is better. I was using `exec` because I only cared about the first match but that was before I was using `^`. But sure, I'll change it

Reply:
> I don't understand what you mean about the leading/trailing spaces.

I meant this:

```
{% something %}
  ^         ^
```

Reply:
They're not mandatory for what I could gather from parser's specs